lib/pubnub/validators/grant_token.rb

```diff
@@ -0,0 +1,63 @@
+# Toplevel Pubnub module.
+module Pubnub
+  # Validator module that holds all validators modules
+  module Validator
+    # Validator for Grant event
+    module GrantToken
+      include CommonValidator
+
+      def validate!
+        validate_keys!
+        validate_ttl!
+        validate_permissions!(@uuids, ":uuids")
+        validate_permissions!(@channels, ":channels")
+        validate_permissions!(@channel_groups, ":uuids")
+        validate_all_permissions!
+      end
+
+      private
+
+      def validate_keys!
+        raise(
+          ArgumentError.new(object: self, message: ':subscribe_key is required for grant token event.'),
+          ':subscribe_key is required for grant token event.'
+        ) if @subscribe_key.nil? || @subscribe_key.empty?
+
+        raise(
+          ArgumentError.new(object: self, message: ':secret_key is required for grant token event.'),
+          ':publish_key is required for grant token event.'
+        ) if @secret_key.nil? || @secret_key.empty?
+      end
+
+      def validate_ttl!
+        return unless !@ttl.nil? && !@ttl.is_a?(Integer)
+
+        raise(
+          ArgumentError.new(object: self, message: ':ttl has to be kind of Integer for grant token event.'),
+          ':ttl has to be kind of Integer for grant token event.'
+        )
+      end
+
+      def validate_permissions!(arg, name)
+        return if arg.nil?
+
+        raise(
+          ArgumentError.new(object: self, message: ":#{name} has to be kind of Hash for grant token event."),
+          ":#{name} has to be kind of Hash for grant token event."
+        ) unless arg.is_a?(Hash)
+      end
+
+      def validate_all_permissions!
```

Comment:
I think SDK should not validate all permissions. Our acceptance tests has scenario to test the request to server and capture response for this case.

Reply:
Yeah. You're probably right

Reply:
Although all of acceptance scenarios are passing, so it's not creating any problems I think

Reply:
so validation not working and request pass-through to server?

Reply:
I'll check it

Reply:
Heh. That's funny. The test is working and the validation is working as well 😆 When we have this scenario that we're not granting anything we're actually sending these grants for one uuid: `Permissions: {:read => false, :write => false, :manage => false , :delete => false, :create => false, :get => false, :update => false, :join => false}`. So... the validation is ok, at least one of `uuids`, `channels` or `channel_groups` are not empty and on the other hand nothing has been granted and we can test the server response

Reply:
I'll remove that validation